core/src/main/java/tech/tablesaw/api/ColumnType.java

```diff
@@ -82,10 +82,13 @@ static ColumnType valueOf(String name) {
   /** TODO: Research this method to provide a good comment */
   AbstractColumnParser<?> customParser(ReadOptions options);
 
-  /** TODO: Research this method to provide a good comment */
-  default boolean compare(int rowNumber, Column<?> temp, Column<?> original) {
-    Object o1 = original.get(rowNumber);
-    Object o2 = temp.get(temp.size() - 1);
+  /**
+   * Compares the row at {@code rowNumber} in {@code column1} and {@code column2} and returns whether they are equal.
+   * @throws {@code IndexOutOfBoundsException} if {@code rowNumber} exceeds either column size
+   */
+  default boolean compare(int rowNumber, Column<?> column1, Column<?> column2) {
```

Comment:
this feels like kind of a weird method honestly. I wonder if we even need it to be a standalone method or if we can just inline it into `compareRows`

Reply:
It might be used by people for other purposes. No harm in keeping it I think.

Reply:
We're changing the functioning of it pretty drastically. Anyone who is relying on it is going to be broken. Better for them to find that out at compile-time because we removed it than at runtime